documentation/typescript/typescript-components-definitions.asciidoc

```diff
@@ -0,0 +1,228 @@
+---
+title: "Appendix: components TypeScript definitions"
+order: 170
+layout: page
+---
+
+== Appendix: components TypeScript definitions
+
+Vaadin components come with TypeScript definitions helping to use web components in TypeScript views.
+The type definitions are `d.ts` files generated every time when new release is published to npm.
+
+Support for TypeScript definitions is added in Vaadin 17. In most of cases, it does not require any
+changes to the code. At the same time, using proper types for the web components helps to make the
+client side views more reliable. Depending on the IDE you use, TypeScript definitions can also give
+additional benefits like better code completion and auto import.
+
+If you are using Visual Studio Code, we recommend to install
+https://marketplace.visualstudio.com/items?itemName=runem.lit-plugin[lit-plugin] which provides
+syntax highlighting and type checking support for LitElement templates. You can also use
+https://www.npmjs.com/package/lit-analyzer[lit-analyzer] CLI tool by the same author for type checking.
+
+== Importing type declarations [[importing-type-declarations]]
+
+Every TypeScript definition file exports type declarations, which describe the public API provided
+by the web component. In order to use the type declarations in your client side views, you need to
+import them using https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-exports[type-only import syntax]:
+
+[source, typescript]
+----
+import type { DialogElement } from '@vaadin/vaadin-dialog';
+----
+
+The type declarations are only used by the TypeScript compiler to statically analyze the code and
+collect information about types. They are removed during `webpack` compilation step and will not end
+up in the resulting JavaScript bundle.
+
+== Using type declarations [[using-type-declarations]]
+
+In some cases you might need to get a reference to the web component, and use it to set properties
+or call methods. Below you will find examples of how to improve developer experience in such cases
+by applying proper types.
+
+=== `@query` decorator [[query-decorator]]
```

Comment:
General recommendation/guideline: avoid using formatted text in headlines. Use plain text.

Formatted text can cause problems in styling the text, as the formatted text might end up being visible in search results, table of contents listings. Formatted text can also use different font size and color than the surrounding plain text, which can make it look visually awkward.

Reply:
Thanks, done.